### ISSUE TITLE
Unable to find credential file in custom config directory

### ISSUE THREAD

**Issue by domanchi** [secrethub/secrethub-cli#339]
## Issue

When I use a custom config directory, I get the error message:

```
Encountered an error: could not find credential file. Run `secrethub signup` to create an account.
```

However, when I specify the credential file directly from the config directory, it seems to work.

## Reproduction Steps

```
# secrethub --version
secrethub version 0.34.0, build 549b444

# echo $SECRETHUB_CONFIG_DIR
/user/config/.secrethub

# secrethub read example/secret
Encountered an error: could not find credential file. Run `secrethub signup` to create an account.
# secrethub --config-dir='/user/config/.secrethub' read example/secret
Encountered an error: could not find credential file. Run `secrethub signup` to create an account. 

# secrethub --credential=`cat $SECRETHUB_CONFIG_DIR/credential` read example/secret
success
```

## Expected Outcome

`secrethub` should respect the config directory settings, and find/read the `credential` file in the config directory.

## Investigation Notes

I tried looking in https://github.com/secrethub/secrethub-go/blob/4b34720c13328528a5db7fc1c588b8e4c2b6b451/pkg/secrethub/configdir/dir.go#L36-L58, and it seems like it should do the expected thing -- so I'm super stumped.

I've also made sure it isn't a UNIX permission issue with the `credential` file, since I used root to perform the reproduction steps.

**Comment by domanchi** [comment]
Oh, super interesting. I was able to fix this issue myself.

**tl;dr: I think the root cause of this issue was that I installed `secrethub-cli` through `snap`. To fix this issue, I just installed it by downloading the right binary from the latest releases on https://github.com/secrethub/secrethub-cli/releases**

----------

For anyone else that stumbles upon this issue in the future, I realized another related, very strange issue when using the `secrethub` binary: **it wouldn't read local files**. For example,

```
# secrethub inject --in-file secrets.yaml
could not read input file: no such file or directory

# cat secrets.yaml
PASSWORD={{ example/secret }}
```

So I reversed why I decided to install it via `snap`, and realized that the [instructions](https://secrethub.io/docs/reference/cli/install/) failed with this error message:

```
N: Skipping acquire of configured file 'main/binary-arm64/Packages' as repository 'https://apt.secrethub.io stable InRelease' doesn't support architecture 'arm64'
```

Huh. How interesting. (FWIW, I'm on a raspberry PI 4, with Ubuntu 20.04)

So, I decided to just download the binaries from the releases, and everything started working! Who would have thought it would be a hardware issue?

**Comment by florisvdg** [comment]
Ah, that makes sense! We [dropped support for Snap in February](https://github.com/secrethub/secrethub-cli/releases/tag/v0.35.0) and [delisted](https://snapcraft.io/search?q=secrethub) SecretHub from the Snap store, exactly because of these types of issues (related to the very restrictive sandbox model that Snap imposes).

On Ubuntu, `apt` would indeed be the most logical option to install the CLI, but you're right in that it doesn't offer `arm` binaries at the moment, so getting the binary from the releases is probably the best option for now. You could also try [Homebrew for Linux](http://linuxbrew.sh): `brew install secrethub/tools/secrethub-cli`. Should support `arm64` too.
